crates/matrix-sdk-crypto/src/session_manager/group_sessions/share_strategy.rs

```diff
@@ -528,6 +616,192 @@ mod tests {
         assert_eq!(code, &WithheldCode::Unauthorised);
     }
 
+    #[async_test]
+    async fn test_share_identity_strategy_no_cross_signing() {
+        let machine: OlmMachine = OlmMachine::new(
+            KeyDistributionTestData::me_id(),
+            KeyDistributionTestData::me_device_id(),
+        )
+        .await;
+
+        let keys_query = KeyDistributionTestData::dan_keys_query_response();
+        let txn_id = TransactionId::new();
+        machine.mark_request_as_sent(&txn_id, &keys_query).await.unwrap();
+
+        let fake_room_id = room_id!("!roomid:localhost");
+
+        let strategy = CollectStrategy::new_identity_based();
+
+        let encryption_settings =
+            EncryptionSettings { sharing_strategy: strategy.clone(), ..Default::default() };
+
+        let request_result = machine
+            .share_room_key(
+                fake_room_id,
+                vec![KeyDistributionTestData::dan_id()].into_iter(),
+                encryption_settings.clone(),
+            )
+            .await;
+
+        assert!(request_result.is_err());
+        let err = request_result.unwrap_err();
+        match err {
+            OlmError::RoomKeySharingStrategyError(
+                crate::error::RoomKeySharingStrategyError::CrossSigningNotSetup,
+            ) => {
+                // ok
+            }
+            _ => panic!("Unexpected share error"),
+        }
```

Comment:
this is better written with `assert_matches`. Something like:

```suggestion
        assert_matches!(
            request_result,
            Err(OlmError::RoomKeySharingStrategyError(RoomKeySharingStrategyError::CrossSigningNotSetup)),
        );
```

Reply:
likewise throughout